File: Easy/27_Remove_Element/Solution.java



public class Solution {
    public int removeElement(int[] nums, int val) {
        int ans = 0;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != val) {
                nums[ans] = nums[i];
                ans++;
            }
        }

        return ans;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        int[] nums1 = {3,2,2,3};
        int val1 = 3;
        System.out.println(solution.removeElement(nums1, val1)); // Output: 2

        // test cases 2
        int[] nums2 = {0,1,2,2,3,0,4,2};
        int val2 = 2;
        System.out.println(solution.removeElement(nums2, val2)); // Output: 5
    }
}

File: Easy/27_Remove_Element/Solution.py
from typing import List, Optional


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        ans = 0

        for i in range(len(nums)):
            if nums[i] != val:
                nums[ans] = nums[i]
                ans += 1

        return ans


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    nums1 = [3, 2, 2, 3]
    val1 = 3
    print(solution.removeElement(nums1, val1))  # Output: 2

    # test cases 2
    nums2 = [0, 1, 2, 2, 3, 0, 4, 2]
    val2 = 2
    print(solution.removeElement(nums2, val2))  # Output: 5